src/indra_cogex/client/queries.py

```diff
@@ -952,3 +982,80 @@ def get_schema_graph(client: Neo4jClient) -> nx.MultiDiGraph:
             label=edge.type,
         )
     return graph
+
+
+# CCLE
+
+
+def is_gene_mutated(
+    client: Neo4jClient, gene: Tuple[str, str], cell_line: Tuple[str, str]
+) -> bool:
+    """Return True if the gene is mutated in the given cell line.
+
+    Parameters
+    ----------
+    client :
+        The Neo4j client.
+    gene :
+        The gene to query.
+    cell_line :
+        The cell line to query.
+
+    Returns
+    -------
+    :
+        True if the gene is mutated in the given cell line.
+    """
+    return client.has_relation(
+        gene,
+        cell_line,
+        relation="mutated_in",
+        source_type="BioEntity",
+        target_type="BioEntity",
+    )
+
+
+# Indra DB
+
+
+def get_drugs_for_target(
+    client: Neo4jClient, target: Tuple[str, str]
+) -> Iterable[Agent]:
+    """Return the drugs targetting the given protein."""
+    rels = client.get_source_relations(
+        target, "indra_rel", source_type="BioEntity", target_type="BioEntity"
+    )
+    drug_rels = [rel for rel in rels if _is_drug_relation(rel)]
+    drug_nodes = [(rel.source_ns, rel.source_id) for rel in drug_rels]
+    stmt_jsons = [json.loads(rel.data["stmt_json"]) for rel in drug_rels]
+    stmts = stmts_from_json(stmt_jsons)
+    drug_agents = [stmt.subj for stmt in stmts]
+    return {"nodes": drug_nodes, "agents": drug_agents}
+
+
+def get_targets_for_drug(client: Neo4jClient, drug: Tuple[str, str]) -> Iterable[Agent]:
+    """Return the proteins targetted by the given drug."""
+    rels = client.get_target_relations(
+        drug, "indra_rel", source_type="BioEntity", target_type="BioEntity"
+    )
+    target_rels = [rel for rel in rels if _is_drug_relation(rel)]
+    target_nodes = [(rel.target_ns, rel.target_id) for rel in target_rels]
+    stmt_jsons = [json.loads(rel.data["stmt_json"]) for rel in target_rels]
+    stmts = stmts_from_json(stmt_jsons)
+    target_agents = [stmt.obj for stmt in stmts]
+    return {"nodes": target_nodes, "agents": target_agents}
+
+
+def is_drug_target(
+    client: Neo4jClient, drug: Tuple[str, str], target: Tuple[str, str]
+) -> bool:
+    """Return True if the drug targets the given protein."""
+    rels = client.get_relations(
+        drug, target, "indra_rel", source_type="BioEntity", target_type="BioEntity"
+    )
+    return any(_is_drug_relation(rel) for rel in rels)
```

Comment:
could be potentially faster to bake the statement type filter into the query itself

Reply:
I'm not sure that's possible with the current API of get_relations (since we don't have a slot for specifying edge property constraints). Since both drug and target are given here, I don't think it will make a big difference.